### PR TITLE
Actionable Notifications

### PR DESCRIPTION
**Pull request by RCGV1**
## What changed?

Creates Actionable notifications for DMs and channel messages. Includes thumbs down/up tapbacks and reply.

## Why did it change?

Useful for quite responses and watch OS.
## How is this tested?

on my iPhone
## Screenshots/Videos (when applicable)
IMG_D3D0EDA4C83E-1

## Checklist

- [ ] My code adheres to the project's coding and style guidelines.
- [ ] I have conducted a self-review of my code.
- [ ] I have commented my code, particularly in complex areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have tested the change to ensure that it works as intended.

### CODE REVIEW

**Comment by RCGV1** [comment]
The new PR

File: Meshtastic/Helpers/BLEManager.swift [hunk 1]
```diff
@@ -3309,9 +3309,9 @@ extension BLEManager: CBCentralManagerDelegate {
 		} else {
 			isSwitchedOn = false
 		}
-
+		
 		var status = ""
-
+
```
Comment (powersjcb) [inline Meshtastic/Helpers/BLEManager.swift:3317]: We should update the linter to keep this consistent across the entire project. (there's some inconsistency for this between files touched by this PR)

File: Meshtastic/Helpers/LocalNotificationManager.swift [hunk 2]
```diff
@@ -5,6 +5,16 @@ import OSLog
...
     var notifications = [Notification]()
+	
+	let thumbsUpAction = UNNotificationAction(identifier: "messageNotification.thumbsUpAction", title:
+													"👍 \(Tapbacks.thumbsUp.description)", options: [])
+		let thumbsDownAction = UNNotificationAction(identifier: "messageNotification.thumbsDownAction", title:
+														"👎  \(Tapbacks.thumbsDown.description)", options: [])
+		let replyInputAction =  UNTextInputNotificationAction(
+					identifier: "messageNotification.replyInputAction",
+					title: "reply".localized,
+					options: [])
```
Comment (powersjcb) [inline Meshtastic/Helpers/LocalNotificationManager.swift:16]: indentation in this diff off. Can we fix this by running the linter?

File: Meshtastic/Helpers/LocalNotificationManager.swift [hunk 3]
```diff
@@ -31,6 +41,15 @@ class LocalNotificationManager {
 
     // This function iterates over the Notification objects in the notifications array and schedules them for delivery in the future
     private func scheduleNotifications() {
+		let messageNotificationCategory = UNNotificationCategory(
+				 identifier: "messageNotificationCategory",
+				 actions: [thumbsUpAction, thumbsDownAction,replyInputAction],
+				 intentIdentifiers: [],
+				 options: .customDismissAction
```
Comment (powersjcb) [inline Meshtastic/Helpers/LocalNotificationManager.swift:48]: ++

File: Meshtastic/MeshtasticAppDelegate.swift [hunk 4]
```diff
@@ -38,14 +38,72 @@ class MeshtasticAppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificat
...
+			} else {
+				Logger.services.error("Failed to retrieve channel or messageId from userInfo")
+			}
+			break
+			
+		case "messageNotification.replyInputAction":
+			if let userInput = (response as? UNTextInputNotificationResponse)?.userText,
+			   let channel = userInfo["channel"] as? Int32,
+			   let replyID = userInfo["messageId"] as? Int64 {
+				let tapbackResponse = !BLEManager.shared.sendMessage(
```
Comment (powersjcb) [inline Meshtastic/MeshtasticAppDelegate.swift:90]: Two questions here:
* It looks like for each of these the variable `tapbackResponse` is never used. Can they be removed?
* Do we need to consider any edge-cases that could cause this to crash? (ie: BLE is disconnected)

File: Meshtastic/Helpers/BLEManager.swift (hunk 1 above)
Comment (powersjcb) [reply Meshtastic/Helpers/BLEManager.swift:3317]: https://github.com/meshtastic/Meshtastic-Apple/pull/919/files

Looks like we haven't been consistently running the linter before merge. (You can probably resolve most of these asks with)


`
$ brew install swiftlint
$ swiftlint --fix <updatedFiles...>
`

File: Meshtastic/MeshtasticAppDelegate.swift (hunk 4 above)
Comment (RCGV1) [reply Meshtastic/MeshtasticAppDelegate.swift:90]: sendMessage returns a boolean if it fails or goes through. For logging purposes we could use it.

**Comment by garthvh** [comment]
Merging into a branch in the repo for testing

File: Meshtastic/Helpers/BLEManager.swift (hunk 1 above)
Comment (garthvh) [reply Meshtastic/Helpers/BLEManager.swift:3317]: Swift lint runs on a GitHub action

**Pull request merged**